fs/operations/check.go

```diff
@@ -374,6 +375,19 @@ func CheckDownload(ctx context.Context, opt *CheckOpt) error {
 	return CheckFn(ctx, &optCopy)
 }
 
+// ApplyTransforms handles --no-unicode-normalization and --ignore-case-sync for CheckSum
```

Comment:
> This doesn't quite implement the same transform as march which also takes into account whether the destination is case sensitive or not - is that important?

I noticed that and almost changed this function to accept a `Fdst` as well, but the reason I didn't is because when it is called from [`ParseSumFile()`](https://github.com/rclone/rclone/blob/c2ac5e4572cab222deb32bb2264fd7be3b4de155/fs/operations/check.go#L579), there is no `Fdst` in scope. This could be changed fairly easily, but would require either changing `ParseSumFile()` to accept a `Fdst` solely for the purpose of passing it to `ApplyTransforms()` (or whatever we end up calling it), or else maybe passing it through the context. Requiring a `Fdst` might also limit its usefulness to other packages. Thoughts?

> Maybe we should be trying to make the march code public instead of re-implementing it here?

Certainly open to that! It would probably mean refactoring the March code a bit, as currently the parts dealing with `transforms` are not in a separate function that could be easily exported. Which approach do you prefer?

Reply:
I don't think we know were we are going with this just yet, so leave the implementation how you have it and we can see how it develops!